src/microsAuthentication/content.md

```diff
@@ -0,0 +1,1075 @@
+
+If your Role grant you access to create and delete Pods, you won't be able to amend Secrets, or create ConfigMaps — for example.
+
+_Could you use Service Accounts as a mechanism to authenticate requests between apps in the cluster?_
+
+Kubernetes offers the same primitives to authentication requests just like Keycloak and Dex.
+
+_What if the Kubernetes API could be used as an Authorisation server?_
+
+_Could you use Service Accounts as a mechanism to authenticate requests between apps in the cluster?_
+
+Let's try that.
+
+You will now deploy two services:
+
+1. You will refer to these services as the API service (API) and the Secret store.
+1. They are written in the Go programming language and they communicate via HTTP.
+1. Each service runs in its own namespaces and use dedicated service accounts.
+1. The secret store requires that any client identifies itself before making a request.
+
+## Creating the cluster
+
+You will need access to a Kubernetes cluster with the [ServiceAccountVolume projection](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection) feature enabled.
+
+Don't worry if you don't know what a ServiceAccountVolume is — you will learn more about it later on in the article.
+
+The ServiceAccountVolume projection requires the Kubernetes API server to be started with certain specific API flags.
+
+The support for different managed Kubernetes providers may vary.
+
+However, you can enable it in [minikube](https://github.com/kubernetes/minikube) with:
+
+```terminal|command=1-6|title=bash
+minikube start \
+  --extra-config=apiserver.service-account-signing-key-file=\
+    /var/lib/minikube/certs/sa.key \
+  --extra-config=apiserver.service-account-issuer=\
+    kubernetes/serviceaccount \
+  --extra-config=apiserver.service-account-api-audiences=api
+```
+
+You should also clone the repository <https://github.com/amitsaha/kubernetes-sa-volume-demo> as it contains the demo source code that will be referred to in the article.
+
+## Deploying the API component
+
+The API service is a headless web application listening on port 8080.
+
+When a client makes any request to it, the API component:
+
+1. Reads the service account token.
+1. Makes a HTTP GET request to the secret store service with the token linked to the Service Account.
+1. Forwards the response.
+
+You can deploy the app in the cluster with:
+
+```terminal|command=1|title=bash
+kubectl apply -f service_accounts/api/deployment.yaml
+namespace/api created
+serviceaccount/api created
+deployment.apps/app created
+```
+
+You can temporarily expose the API with:
+
+```terminal|command=1|title=bash
+kubectl -n api expose deployment
+service/app exposed
+```
+
+Retrieve the URL of the app with:
+
+```terminal|command=1|title=bash
+minikube service deployment --url
+TODO: missing output
+```
+
+_Will it work?_
+
+You can issue a request with:
+
+```terminal|command=1|title=bash
+curl <insert URL from above>
```

Comment:
If we don't amend the Service earlier, this should container an example of the output above instead of `<>`